source/slang/slang-emit-spirv.cpp

```diff
@@ -2730,6 +2739,40 @@ struct SPIRVEmitContext
                 auto inner = ensureInst(location);
                 registerInst(inst, inner);
                 result = inner;
+                break;
+            }
+        case kIROp_RequireComputeDerivative:
+            {
+                auto parentFunc = getParentFunc(inst);
+                if (hasExtensionDeclaration(UnownedStringSlice("SPV_NV_compute_shader_derivatives")))
+                    break;
+
+                HashSet<IRFunc*>* entryPointsUsingInst = EntryPointsUsingInst(*this, parentFunc);
+                for (IRFunc* entryPoint : *entryPointsUsingInst)
+                {
+                    auto entryPointDecor = entryPoint->findDecoration<IREntryPointDecoration>();
+                    if (!entryPointDecor || entryPointDecor->getProfile().getStage() != Stage::Compute)
+                        continue;
+
+                    ensureExtensionDeclaration(UnownedStringSlice("SPV_NV_compute_shader_derivatives"));
```

Comment:
Does this mean that if I have a module with two compute shaders, the execution mode will only get declared for the first compute shader?